pagopa/io-app
# chore: [IAI-106] Upgrade ESLint toolchain

## Short description
As noted [in this PR](https://github.com/pagopa/io-app/pull/3646#issuecomment-1010849106) there are some dependencies connected to ESLint that can be easily upgraded.

## List of changes proposed in this pull request
- Upgrade the relevant packages to the latest available version
- Fix one linting error related to the new [no-unnecessary-type-constraint](https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/docs/rules/no-unnecessary-type-constraint.md) rule

## How to test
Being it a toolchain-only update, a green CI should be enough.

**Please Note** that due to the large number of packages involved, a clean install is needed.
